domains/pallets/messenger/src/lib.rs

```diff
@@ -1069,15 +1094,18 @@ mod pallet {
                 {
                     // channel is being opened without an owner since this is a relay message
                     // from other chain
-                    Self::do_init_channel(msg.src_chain_id, params, None).map_err(|err| {
-                        log::error!(
-                            "Error initiating channel: {:?} with chain: {:?}: {:?}",
-                            msg.channel_id,
-                            msg.src_chain_id,
-                            err
-                        );
-                        InvalidTransaction::Call
-                    })?;
+                    // we do not check the allowlist to finish the end to end flow
+                    Self::do_init_channel(msg.src_chain_id, params, None, false).map_err(
```

Comment:
I'm confused, is "the end to end flow" mean even if the src chain is not in the allowlist the dst chain still accepts the `init_channel` req and then waits for the src chain to close the channel manually?

IMHO, in this case, ideally the dst chain should return an error XDM response and the src chain closes the init status channel when processing that response, so there is no need to close the init status channel manually. Or maybe I missed something.

Reply:
Right now, dst_chain will return the Err through XDM and src_chain will not open the channel. I have let the closing channel to be a manual process since there is a reserve fee and channel initiator will have the need to unlock their funds. This ensures they not only pay the portion of reserve fee but also needs to pay for the closing channel fee as well. If the src_chain closing it automatically, they no one is paying for it instead

Reply:
Yes that is the current behavior. What Am i missing ?

Reply:
I mean if the XDM resp tx fee is a problem then it also exists in the endpoint XDM, if we follow the same approach as the close channel XDM to fix it, we will need the user manually submit extrinsic to get refund, which is obvious a bad UX. We should consider a better solution like pre-charge the XDM resp tx fee when the user send the close channel/transporter XDM.

Reply:
I'm not following what you mean here unfortunately

Reply:
From your [comment](https://github.com/subspace/subspace/pull/2829#discussion_r1632550918) the reason we need to manually close channel here is we want the XDM response tx fee to get paid, but the same situation exists in the XDM response of `transporter` and to collect the tx fee of the XDM response of `transporter` we will require the user manually submit extrinsic to handle the `transporter` XDM response, which is obvious a bad UX. So I proposed other solutions like pre-charge the XDM resp tx fee when the user sends the close channel/transporter XDM.

Reply:
no, that is not the case. Transporter automatically reverts when it receives the err from dst_chain and fees are already acounted for. There is a test case for this specifically. For Closing channel, user needs to manually submit the close request request

Reply:
> For Closing channel, user needs to manually submit the close request request

There are 2 scenarios:
1. The channel is open before the src chain is removed from the allowlist
    - In this scenario, the user does need to manually submit extrinsic to close the channel as we agreed in #2815 
2. The src chain is removed from the allowlist (or the src chain is never added to the allowlist at all) and then an open channel request comes
    - In this PR, even if the src chain is not in the allowlist, the dst chain will still accept the request and init the channel in its state then wait for the src chain to close the channel manually

My question is for scenario 2 why "For Closing channel, user needs to manually submit the close request request" is necessary? if it is all about the XDM response tx fee it can be handled just like the transporter i.e. the dst chain rejected the open channel request and returns an error XDM response, so the channel is never initialized is the dst chain.

Reply:
if we keep disregard the message then relayer would end up trying for however long the message exisits on source chain. Instead we init the channel once and no messages are accepted until user submits the close channel and cleanups all the storage. I like the second option because its completes the flow and relayer can move on

Reply:
>if we keep disregard the message then relayer would end up trying for however long the message exisits on source chain.

Okay, it makes sense in this perspective